### PR TITLE
JI-3263 Add support of report when content is used as child content

### PR DESCRIPTION
**Pull request by ravimajithia**

### CODE REVIEW

File: image-hotspot-question.js [hunk 1]
```diff
@@ -338,6 +347,58 @@ H5P.ImageHotspotQuestion = (function ($, Question) {
...
+    // Individual report not supported
+    if (this.isRoot()) {
+      return;
+    }
+    var definition = {};
+    definition.description = {
+      'en-US': this.getTitle()
+    };
+    definition.type = 'http://adlnet.gov/expapi/activities/cmi.interaction';
+    definition.interactionType = 'general';
```
Comment (otacke) [inline image-hotspot-question.js:398]: I don't think that's a valid value for `interactionType`: https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#interaction-activities. What about `other`?

File: image-hotspot-question.js [hunk 2]
```diff
@@ -371,6 +432,11 @@ H5P.ImageHotspotQuestion = (function ($, Question) {
 
     // Clear feedback
     this.removeFeedback();
+
+    // Reset timer
+    if (this.isRoot()) {
+      this.setActivityStarted(true);
```
Comment (otacke) [inline image-hotspot-question.js:438]: Good thought! Other content types don't do this however, so I'm not sure if this should be introduced here.

File: image-hotspot-question.js [hunk 3]
```diff
@@ -338,6 +347,58 @@ H5P.ImageHotspotQuestion = (function ($, Question) {
...
+  };
+
+  /**
+   * Generate xAPI object definition used in xAPI statements.
+   * @return {Object}
+   */
+  ImageHotspotQuestion.prototype.getxAPIDefinition = function () {
+    // Individual report not supported
+    if (this.isRoot()) {
+      return;
```
[the remark this answers is not in the document]
Comment (otacke) [reply image-hotspot-question.js:391]: Sorry! Forget this. Confused this with `getXAPIData`.

File: image-hotspot-question.js (hunk 1 above)
Comment (ravimajithia) [reply image-hotspot-question.js:398]: Thanks, I will take care in next commit.

File: image-hotspot-question.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ravimajithia) [reply image-hotspot-question.js:391]: I am not sure that I understood this entirely. I am not able to find `getxAPIDefinition` in here - https://h5p.org/documentation/developers/contracts

File: image-hotspot-question.js (hunk 2 above)
Comment (ravimajithia) [reply image-hotspot-question.js:438]: True, It does not seem to require here!

File: image-hotspot-question.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (otacke) [reply image-hotspot-question.js:391]: That's why I said "forget this" = "ignore this comment" :-D My bad. I confused `getxAPIDefinition` with `getXAPIData`.

**Pull request merged**